pkg/controller/service/servicecontroller_test.go

```diff
@@ -484,54 +450,58 @@ func TestSyncService_new(t *testing.T) {
 			testName: "if valid service",
 			key:      "external-balancer",
 			updateFn: func() {
+				testSvc := defaultExternalService()
 				controller, cloud, _ = newController()
-				srv := controller.cache.getOrCreate("external-balancer")
-				srv.state = defaultExternalService()
+				controller.enqueueService(testSvc)
+				svc := controller.cache.getOrCreate("external-balancer")
+				svc.state = testSvc
 			},
 			expectedErr: nil,
 		},
 	}
 
-	for _, tst := range testCases {
-		var printError bool
-		tst.updateFn()
-		obtainedErr := controller.syncService(tst.key)
-		if obtainedErr != tst.expectedErr {
-			printError = true
-			if obtainedErr != nil && tst.expectedErr != nil {
-				if obtainedErr.Error() == tst.expectedErr.Error() {
-					//Check if the actual error matches, if yes then test PASSed
-					printError = false
-				}
-			}
+	for _, tc := range testCases {
+
+		tc.updateFn()
+		obtainedErr := controller.syncService(tc.key)
+
+		//Lets only check if we expect error but recived nil and viceversa.
+		if (obtainedErr != nil && tc.expectedErr == nil) && (obtainedErr == nil && tc.expectedErr != nil) {
+			t.Errorf("%v Expected=%v Obtained=%v", tc.testName, tc.expectedErr, obtainedErr)
 		}
-		if printError {
-			t.Errorf("%v Expected=%v Obtained=%v", tst.testName, tst.expectedErr, obtainedErr)
+
+		//Post processing the element should not be there in teh sync queue
+		_, exist := controller.cache.get(tc.key)
+		if exist {
+			t.Fatalf("sync service error, workingQueue should not contain service: %s any more", tc.key)
 		}
 	}
+
 }
 
-func TestDeleteService(t *testing.T) {
+func TestProcessDeleteService(t *testing.T) {
```

Comment:
Make it TestProcessServiceDeletion to be consistent with the origin function name.